tests/fallback/fallback_test.go

```diff
@@ -22,6 +25,75 @@ var _ = Describe("cOS booting fallback tests", func() {
 			s.Reset()
 		}
 	})
+
+	Context("image is corrupted", func() {
+		BeforeEach(func() {
+			s.Snapshot()
+		})
+
+		AfterEach(func() {
+			if CurrentGinkgoTestDescription().Failed {
+				s.GatherAllLogs()
+			}
+			if CurrentGinkgoTestDescription().Failed == false {
+				s.RestoreSnapshot()
+			}
+		})
+
+		It("boots in fallback when rootfs is damaged", func() {
+			currentVersion := s.GetOSRelease("VERSION")
+
+			// Auto assessment was installed
+			out, _ := s.Command("sudo cat /run/initramfs/cos-state/grubcustom")
+			Expect(out).To(ContainSubstring("bootfile_loc"))
+
+			out, _ = s.Command("sudo cat /run/initramfs/cos-state/grub_boot_assessment")
+			Expect(out).To(ContainSubstring("boot_assessment_blk"))
+
+			cmdline, _ := s.Command("sudo cat /proc/cmdline")
+			Expect(cmdline).To(ContainSubstring("rd.emergency=reboot rd.shell=0 panic=5"))
+
+			out, err := s.Command(fmt.Sprintf("elemental upgrade --no-verify --docker-image %s:cos-system-%s", s.GreenRepo, s.TestVersion))
+			Expect(err).ToNot(HaveOccurred(), out)
+			Expect(out).Should(ContainSubstring("Upgrade completed"))
+			Expect(out).Should(ContainSubstring("Upgrading active partition"))
+
+			out, _ = s.Command("sudo cat /run/initramfs/cos-state/boot_assessment")
+			Expect(out).To(ContainSubstring("osupgrade=done"))
+
+			// Break the upgrade
+			out, _ = s.Command("sudo mount -o rw,remount /run/initramfs/cos-state")
+			fmt.Println(out)
+
+			out, _ = s.Command("sudo mkdir -p /tmp/mnt/STATE")
+			fmt.Println(out)
+
+			s.Command("sudo mount /run/initramfs/cos-state/cOS/active.img /tmp/mnt/STATE")
+
+			for _, d := range []string{"usr/lib/systemd"} {
```

Comment:
here we trigger boot assessment by deleting systemd from the image after the upgrade was done

Reply:
NOTE: triggers a kernel panic

Reply:
this is weird. on c3os it triggered a kernel panic while on cOS it just makes systemd go into freeze mode. Anyhow, I've added `systemd.crash_reboot` so it does restart when PID 1 goes into freeze infinite loops.